pcstac/pcstac/client.py

```diff
@@ -84,10 +92,14 @@ async def all_collections(self, **kwargs: Dict[str, Any]) -> Collections:
         Returns:
             Collections.
         """
-        try:
-            return collections_endpoint_cache["/collections"]
-        except KeyError:
-            collections = await super().all_collections(**kwargs)
+        await rate_limit(
```

Comment:
Is it possible to convert this to a decorator that could be applied to these functions? Could be nice from a portability/visibility perspective.

Reply:
Good call - I did this for rate_limit and also the new back_pressure.